app/src/main/java/eu/laramartin/medsreminder/reminders/NotificationUtils.java

```diff
@@ -0,0 +1,106 @@
+//                context.getSystemService(Context.NOTIFICATION_SERVICE);
+//        notificationManager.cancelAll();
+//    }
+
+    public static void remindUserToTakeMed(Context context) {
+
+        NotificationCompat.Builder notificationBuilder = new NotificationCompat.Builder(context)
+                .setColor(ContextCompat.getColor(context, R.color.colorPrimary))
+                .setSmallIcon(R.drawable.ic_capsule_128)
+                .setLargeIcon(largeIcon(context))
+                .setContentTitle(context.getString(R.string.its_time_reminder_title))
+                .setContentText(context.getString(R.string.med_reminder_notification_text))
+                .setStyle(new NotificationCompat.BigTextStyle().bigText(
+                        context.getString(R.string.med_reminder_notification_text)))
+                .setDefaults(Notification.DEFAULT_VIBRATE)
+                .setContentIntent(contentIntent(context))
+                .addAction(takeMedAction(context))
+//                .addAction(ignoreReminderAction(context))
+                .setAutoCancel(true);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+            notificationBuilder.setPriority(Notification.PRIORITY_HIGH);
+        }
+
+        NotificationManager notificationManager = (NotificationManager)
+                context.getSystemService(Context.NOTIFICATION_SERVICE);
+
+        /* WATER_REMINDER_NOTIFICATION_ID allows you to update or cancel the notification later on */
+        notificationManager.notify(MED_REMINDER_NOTIFICATION_ID, notificationBuilder.build());
+    }
+
+//    private static NotificationCompat.Action ignoreReminderAction(Context context) {
+//        Intent ignoreReminderIntent = new Intent(context, ReminderIntentService.class);
+//        ignoreReminderIntent.setAction(ReminderTasks.ACTION_DISMISS_NOTIFICATION);
+//        PendingIntent ignoreReminderPendingIntent = PendingIntent.getService(
+//                context,
+//                ACTION_IGNORE_PENDING_INTENT_ID,
+//                ignoreReminderIntent,
+//                PendingIntent.FLAG_UPDATE_CURRENT);
+//        NotificationCompat.Action ignoreReminderAction = new NotificationCompat.Action(R.drawable.ic_cancel_black_24px,
+//                "No, thanks.",
+//                ignoreReminderPendingIntent);
+//        return ignoreReminderAction;
+//    }
+
+    static NotificationCompat.Action takeMedAction(Context context) {
+        Intent takeMedIntent = new Intent(context, ReminderIntentService.class);
+        PendingIntent takeMedPendingIntent = PendingIntent.getService(
+                context,
+                ACTION_TAKE_MED_PENDING_INTENT_ID,
+                takeMedIntent,
+                PendingIntent.FLAG_CANCEL_CURRENT);
+        NotificationCompat.Action takeMedAction = new NotificationCompat.Action(R.drawable.ic_capsule_128,
+                "I did it!",
+                takeMedPendingIntent);
+        return takeMedAction;
+    }
```

Comment:
Add a TODO or similar, since this code is not yet doing the right thing (adding a report on firebase)